pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1090,12 +1090,31 @@ func (m *ManagerImpl) isDevicePluginResource(resource string) bool {
 	return false
 }
 
-// GetAllocatableDevices returns information about all the devices known to the manager
 func (m *ManagerImpl) GetAllocatableDevices() ResourceDeviceInstances {
 	m.mutex.Lock()
-	resp := m.allDevices.Clone()
+	allDevs := m.allDevices.Clone()
```

Comment:
this change use NewResourceDeviceInstances() for response, so allDevs don't need clone

Reply:
The intent is to prevent changes to `allDevices` when the for loop below runs. Creating a clone ensures that this function will return a consistent view of the kubelet devices.

Reply:
the mutex can prevent that so you should not do extra things

Reply:
But this will make the critical section longer